[v10.2.x] Exemplars: Add exemplar configuration to Options Pane to order, hide or unhide exemplar labels on tooltip.

aishyandapalli opened the pull request:
This PR adds a new feature to the Options Pane for PanelEditor. It adds a new section called Exemplar Configuration where you can order, hide or unhide different exemplar labels.

This feature is needed because when there are too many labels for Exemplars, the tooltip is very hard to be readable and we want to provide a configuration for multiple users where they can configure by themselves.

iainfinlayson committed:
Update _index.md (#72222)

The existing link will take new users to the Data Source Management page which falls under that administration section and is perhaps more advanced that we want for a new user. The updated link takes new users to what I think was the intended "Data Sources" page.

sympatheticmoose committed:
docs: remove broken plugin from Angular guidance (#72614)

aceiot-svg-panel fails to load in G10 and is therefore (currently) an unviable alternative

codeincarnate committed:
Transformations: Format Time Fixes (#72571)

* Add docs

* Use interactive tooltip

joshhunt committed:
Chore: Remove eval source maps to fix trusted types during dev (#72664)

* user essentials mob! :trident:

lastFile:scripts/webpack/webpack.dev.js

* user essentials mob! :trident:

* delete results.md

Co-authored-by: Joao Silva <joao.silva@grafana.com>
Co-authored-by: Ashley Harrison <ashley.harrison@grafana.com>
Co-authored-by: L-M-K-B <48948963+L-M-K-B@users.noreply.github.com>
Co-authored-by: tskarhed <1438972+tskarhed@users.noreply.github.com>
Co-authored-by: Roxana Turc <anamaria-roxana.turc@grafana.com>

---------

Co-authored-by: Ashley Harrison <ashley.harrison@grafana.com>
Co-authored-by: Joao Silva <joao.silva@grafana.com>
Co-authored-by: L-M-K-B <48948963+L-M-K-B@users.noreply.github.com>
Co-authored-by: tskarhed <1438972+tskarhed@users.noreply.github.com>
Co-authored-by: Roxana Turc <anamaria-roxana.turc@grafana.com>

papagian committed:
Search v1: Remove unnecessary subqueries (#72388)

* Add feature flag

* Introduce interface and dummy implementation

* Add tests for the new filter

* accessControlDashboardPermissionFilterNoFolderSubquery implementation

* join only if it's necessary

* force ordering for tests

* Temporarily enable new query for benchmarks

Jguer committed:
Auth: Move Team service to SignedInUserInterface (#72674)

* move SignedInUser to specific file

* add primitive interface for signedInUser

aangelisc committed:
AzureMonitor: Add hardcoded namespaces to improve coverage (#72629)

Add hardcoded resources

ifrost committed:
Explore: Fix parsing absolute range when URL changes (#72578)

* Fix parsing absolute range when URL changes

* Update public/app/features/explore/hooks/useStateSync/index.test.tsx

Co-authored-by: Haris Rozajac <58232930+harisrozajac@users.noreply.github.com>

* Clean up docs

* Clean up

---------

Co-authored-by: Haris Rozajac <58232930+harisrozajac@users.noreply.github.com>

ashharrison90 committed:
Nested folders: Add e2e test for new browse page (#72590)

* scaffold nested folder structure

* update structure slightly

* update comments

* Added basic scafold for Cypress test

* fix selector name, add table body selector, improve expand aria label

* skip test for now

* undo changes to cypress custom.ini

---------

Co-authored-by: Roxana Turc <anamaria-roxana.turc@grafana.com>

ashharrison90 committed:
Nested folders: Clear selection state in tree view when indeterminate (#72595)

* clear selection state when indeterminate

* ensure search state is properly cleared when toggling the indeterminate checkbox

* select everything in view

papagian committed:
Nested folders: Fix search query for empty self-contained permissions (#72727)

* Add tests

* Fix query for nested folders with zero self-contained permissions

* Fix query behind  permissionsFilterRemoveSubquery flag

* Apply suggestion from code review

RoxanaAnamariaTurc committed:
GrafanaUI: Added spacing token to Divider Component (#72725)

* GrafanaUI: Added spacing token to Divider Component

* Made changes suggested

jsoref committed:
Only run workflows if they might be able to work (#72503)

* Fix whitespace

* Only run workflows if they might be able to work

* fix quotes

---------

Co-authored-by: joshhunt <josh@trtr.co>

eledobleefe committed:
GLDS: Move Text component from the `unstable` package to `grafana-ui` (#72660)

toddtreece committed:
Chore: Move backgroundsvcs and usagestatssvcs to registry (#72692)

aocenas committed:
Flamegraph: Remove feature flag and code to switch to pyroscope version (#72689)

joshhunt committed:
Chore: Fix pr-checks not having enough permissions (#72753)

* Chore: Fix pr-checks not having enough permissions

* update permissions

joshhunt committed:
Chore: Fix pr-checks not having enough permissions (#72754)

itsmylife committed:
InfluxDB: SQL Support (#72167)

* Add influxdbSqlSupport feature toggle

* Add SQL option to the config page

* Add SQL backend

* Add metadata support in config page

* Implement unified querying

* Fix healthcheck query

* fsql tests

* secure grpc by default

* code cleanup

* Query handing for sql mode

* Implement a placeholder sql editor

* Fix query language dropdown

* go mod updates

* make lint-go

* more make lint-go

* remove unused runQuery

* switch statements with default case

* linting again

wbrowne committed:
Plugins: Add inititialization stage to plugin loader pipeline (#72667)

* first pass

* migrate tests

* simplify

* fix comments

* fix linter

* nil checks

* remove comment

matyax committed:
Logs: Improved keyboard accessibility of log rows menu (#72686)

* Log Row Menu Cell: improve accessibility

* Explain implementation in comments

* Make onBlur mandatory

itsmylife committed:
InfluxDB: SQL Query Editor (#72168)

* Add influxdbSqlSupport feature toggle

* Add SQL option to the config page

* Add SQL backend

* Add metadata support in config page

* Implement unified querying

* Fix healthcheck query

* fsql tests

* secure grpc by default

* code cleanup

* Query handing for sql mode

* Implement a placeholder sql editor

* Fix query language dropdown

* drop in SQL editor

* switch to use rawSql, get sql editor working

* fix healthcheck

* WIP

* memoize component to stop unwanted rerender onQuery

* dont reinit datasource on each render of the editor

* remove useless memo

* clean up

* Fix the link

* Alpha state warning

* Remove console.logs

* update model for fsql

* remove unused

---------

Co-authored-by: Galen <galen.kistler@grafana.com>

IevaVasiljeva committed:
Auth: add a lock message for Grafana Admin role (#72724)

* add a lock message for Grafana admin role sync

* Update public/app/features/admin/UserAdminPage.tsx

Co-authored-by: Misi <mgyongyosi@users.noreply.github.com>

* linting

* use theme for spacing

---------

Co-authored-by: Misi <mgyongyosi@users.noreply.github.com>

aishyandapalli committed:
Exemplar label configuration

aishyandapalli committed:
Show all labels if configured exemplars are empty

moxious committed:
Update start-restart-grafana.md (#72737)

* Update start-restart-grafana.md

Closes https://github.com/grafana/grafana/issues/72309

* Update docs/sources/setup-grafana/start-restart-grafana.md

Co-authored-by: Christopher Moyer <35463610+chri2547@users.noreply.github.com>

---------

Co-authored-by: Christopher Moyer <35463610+chri2547@users.noreply.github.com>

tonypowa committed:
Docs update: /docs/sources/developers/http_api/user.md (#72430)

updated endpoint authentication

kminehart committed:
CI: Update PR patch check workflow (#72770)

* Update PR patch check workflow

* trigger ci

* trigger ci

ryantxu committed:
UserSync: Avoid UpdateLastSeenAt with invalid user ids (#72776)

* avoid user zero

* more errors

* more tests

* split

joey-grafana committed:
Tempo: Only get span_names when they are in the response (#72663)

* Only get span names when they are in the response

* Find span_name field/labels

mgyongyosi committed:
Auth: Add key_id config param to auth.jwt (#72711)

* Specify keyID for public key provided in PEM format for JWT Auth

* Update docs

* Update sample.ini

mgyongyosi committed:
Auth: Don't rotate session cookie from frontend for JWT Auth (#72744)

Skip client token rotation for JWT auth

papagian committed:
Usage stats: Tune collector execution startup and interval (#72688)

* Do not update statistics at service collector startup

* Configurable collector interval

* Introduce initial random delay

* Prevent reporting metrics until the stats have been collected

* Apply suggestion from code review

Jguer committed:
Auth: Move Support Bundle service to SignedInUser interface (#72723)

add identity interface for support bundles service

tonypowa committed:
Alerting: added description to api resource (#70597)

* added description to api resource

* prettier now

zerok committed:
CI: Remove pr-commands-closed workflow as replaced with auto-milestone (#72793)

Jguer committed:
Auth: Move Org service to SignedInUser interface (#72720)

add identity interface for org service

itsmylife committed:
Fix setting oauthPassThru flag (#72472)

* Fix setting oauthPassThru flag

* Call onChange only if azure auth is enabled

* Move changes in onSettingsChange

RoxanaAnamariaTurc committed:
A11y: Added aria-label to ColorPicker component (#72666)

* A11y: Added aria-label to ColorPicker component

* Updated the code with changes suggested

cmo-pomerium committed:
Update _index.md (#72317)

* Update _index.md

Adding popular open-source proxy options so users don't have to do further research if they don't want to

* ran Prettier

* Update docs/sources/setup-grafana/configure-security/_index.md

Co-authored-by: Ieva <vasiljeva.ieva@gmail.com>

---------

Co-authored-by: Christopher Moyer <35463610+chri2547@users.noreply.github.com>
Co-authored-by: Ieva <vasiljeva.ieva@gmail.com>

fabrizio-grafana committed:
Fix typos in Loki template variable tests (#72721)

eledobleefe committed:
GLDS: Make Text component story not internal (#72803)

xnyo committed:
Plugins: Add elementId attribute to angular_deprecation_docs_clicked interaction (#72800)

Add elementId attribute to angular_deprecation_docs_clicked interaction

stephaniehingtgen committed:
Proxy: Set the proxy information in Grafana for Graphite (#72777)

eleijonmarck committed:
Auth: Add missing skipOrgRoleSync property to SocialBase for GenericOAuth (#72752)

* add missing cfg for skiporgrolesync

* add google skiporgrolesync

* removed google skip

* update docs to reflect google

* remove docs update for google

alexanderzobnin committed:
Grafana-ui: Fix RadioButtonList component (#72804)

* Grafana-ui: Fix RadioButtonList component

* Use just an index as a key

linoman committed:
Auth: Add `No basic role` UI (#72561)

* Render new No Basic Role

* Add Tooltip to dropdown menu

toddtreece committed:
Chore: Revert dskit service additions (#72608)

polibb committed:
Dashboard: Add analytics events for sharing a dashboard  (#72613)

* activeTab for public dashboard matches others

* add analytics action for going through categories of sharing modal

* add analytics for going through actions of any cateogry in sharing modal

* change 'tab' to 'category'; add key 'sharing_category' to actions for easier distinction

* sharedCategory sounds like a category is shared; changed naming to shareDashboardType

* remove hard-coded analytic event name

* wrap the reportInteraction call for all actions when sharing dashboard

---------

Co-authored-by: Alexandra Vargas <alexa1866@gmail.com>

usmangt committed:
Docs/k8s install update (#72567)

* added before you begin section

* just syncing up

* added namespace and yaml mainfest sections

* started the managed k8s section

* added external-ip section

* added minikube instructions

* added deployment rollout section

* added rolling update section

* added rollback stratgety section

* added the logging section

* fixed many typos

* Update docs/sources/setup-grafana/installation/kubernetes/index.md

* Update docs/sources/setup-grafana/installation/kubernetes/index.md

* Update docs/sources/setup-grafana/installation/kubernetes/index.md

* Update docs/sources/setup-grafana/installation/kubernetes/index.md

* Update docs/sources/setup-grafana/installation/kubernetes/index.md

* adjust codeblock spacing

* adjusts steps

* copy edit throughout

* Apply suggestions from code review

* more copy edit, updates links

* fixed minor typos

* Apply suggestions from code review

Co-authored-by: Clayton Cornell <131809008+clayton-cornell@users.noreply.github.com>

* Apply suggestions from code review

* Apply suggestions from code review

* explained the dry-run example

* Move .codespellignore out of Drone so that technical writers can modify it

Signed-off-by: Jack Baldry <jack.baldry@grafana.com>

* Fix spelling errors

Signed-off-by: Jack Baldry <jack.baldry@grafana.com>

* Add .codespellignore CODEOWNER

Signed-off-by: Jack Baldry <jack.baldry@grafana.com>

---------

Signed-off-by: Jack Baldry <jack.baldry@grafana.com>
Co-authored-by: Christopher Moyer <35463610+chri2547@users.noreply.github.com>
Co-authored-by: Chris Moyer <chris.moyer@grafana.com>
Co-authored-by: Clayton Cornell <131809008+clayton-cornell@users.noreply.github.com>
Co-authored-by: Jack Baldry <jack.baldry@grafana.com>

ivanahuckova committed:
Loki: Introduce `$__auto` range variable for metric queries (#72690)

* Loki: Add  interpolation to backend

* Loki: Replace default variable  with  in frontend

* Loki: Update docs in query builder fro __auto

* Loki: Update test for change default __auto

* Loki: Remove  and  from suggestions as  should be used

* Update docs

* Update pkg/tsdb/loki/parse_query.go

* Fix backend lint

* Fix lint and test

* Update

* Update docs/sources/datasources/loki/template-variables/index.md

Co-authored-by: lwandz13 <126723338+lwandz13@users.noreply.github.com>

* Update public/app/plugins/datasource/loki/querybuilder/operationUtils.ts

Co-authored-by: Matias Chomicki <matyax@gmail.com>

---------

Co-authored-by: lwandz13 <126723338+lwandz13@users.noreply.github.com>
Co-authored-by: Matias Chomicki <matyax@gmail.com>

eleijonmarck committed:
SAML: Add onclick action to providercard if the provider is configured (#72734)

* adds missign onclick for provider card

* simplified version

IevaVasiljeva committed:
Auth: lock down Grafana admin sync for SAML (#72828)

lock down Grafana admin sync for SAML

zuchka committed:
Dashboards: Add megawatt hour (MWh) unit (#72779)

* add megawatt hour MWh unit

* uppercase MWh

idastambuk committed:
Cloudwatch Logs: Set Alerting timeout to datasource config's logsTimeout (#72611)

danielthedm committed:
Updating codeowners for User Service (#72748)

reassigning user service to authnz

IbrahimCSAE committed:
Feat: Feature toggle admin page frontend interface (#72164)

* feature toggles admin page proto

* feature toggle admin page proto

* keep phase 1 code only

* latest update with api

* fix

* fix

* add correct premissions in admin.go

* move behind toggle

* Use InteractiveTable

* guard behind feature toggle

* use RTK

* route in api.go

* fixes

kylebrandt committed:
SSE: Warn on dropped items in Union in Math Operation (#72682)

deresolution20 committed:
Update index.md (#71680)

* Update index.md

Updated the notes section per https://github.com/grafana/support-escalations/issues/6202

* Update docs/sources/setup-grafana/configure-grafana/configure-custom-branding/index.md

---------

Co-authored-by: Christopher Moyer <35463610+chri2547@users.noreply.github.com>

adela-almasan committed:
Units: Fix display for kHz, MHz, GHz(#72845)

fix unit display

aishyandapalli committed:
empty array fix

aishyandapalli committed:
undefined exemplar labels fix

bossinc committed:
AzureMonitor: Check to see if response is nil to prevent panic (#72705)

* check to see if response is nil to prevent panic

* make function more go

* logging error and moving on

* lint

danielthedm committed:
updating codeowners for correlations (#72743)

mikkancso committed:
Admin: Add redirect notice for datasources to admin landing page (#72736)

* update copy of ConnectionsRedirectNotice

The Connections page is not that new anymore

* extend NavLandingPage with optional header

* show ConnectionsRedirectNotice on Admin landing page

* make ConnectionsRedirectNotice dismissable

* make ConnectionsRedirectNotice informational

sjd210 committed:
SQL: Remove double quotes for multi-value variables (#71130)

* RemoveFirstLastQuotes

* MoveTemplateHandleToFront

* PreserveTemplateInQuery

* Revert "RemoveFirstLastQuotes"

This reverts commit e1ef5764b896084d9bdf2564c6f4d1d0fd76db07.

* CheckForTemplateVarType

* CheckTemplateVars

* ReRunChecks

* ExtractMultiTestToFunction + AddTests

* RemoveAsyncFromTests

* Move condition definition out of .some()

kalleep committed:
Authn: client token rotation schedule (#72809)

* ClientTokenRotation: Only rotate the token if we have a session expiry time

* ContextSrv: Remove guard

linoman committed:
Auth: Add no role frontend feature flag (#72823)

* Add 'noBasicRole' feature flag

* Hide role options and tooltip with feature flag

* Add feature flag to registry

joshhunt committed:
Search API: Search by folder UID (#65040)

* Search: Attempt to support folderUID filter

* Search: Use folder UID instead of ID for searching folders

* Update swagger

* Fix JSON property casing

* Add integration test

* Remove redundant query condition

* Fix frontend test

* Fix listing dashboards in General/root

* Add support for fetching top level folders

using `folderUIDs=` (empty string) query parameter

* Add deprecation notice

* Send uid of general in sql.ts

* Use 'general' for query folderUIDs query param for fetching folder

* Add tests

* Fix FolderUIDFilter

---------

Co-authored-by: Sofia Papagiannaki <1632407+papagian@users.noreply.github.com>

wbrowne committed:
Plugins: Add termination stage to plugin loader pipeline (#72822)

* add termination stage

* uid -> pluginID (for now)

* also fix fakes

* add simple test

* Fix logger name

Co-authored-by: Giuseppe Guerra <giuseppe.guerra@grafana.com>

* inline stop func call

Co-authored-by: Giuseppe Guerra <giuseppe.guerra@grafana.com>

---------

Co-authored-by: Giuseppe Guerra <giuseppe.guerra@grafana.com>

wbrowne committed:
Plugins: Add remaining steps to inititialization stage (#72894)

add remainder of init steps

mildwonkey committed:
chore: minor refactoring in the grafana server cli command (#72847)

This is preparative work for extending the grafana server cli command to target individual dskit modules that seemed worth breaking into this smaller PR. This moves the CLI flags and various reusable chunks of code into variables and methods.

moxious committed:
Update bind-net-capabilities.md (#65181)

This doc fragment is included on multiple pages including this one: https://grafana.com/docs/grafana/latest/setup-grafana/installation/debian/

the only flaw is it's missing "sudo".  Modern linux practice suggests using sudo for all administrative edits like this, and this change reinforces parallel structure in the target docs page.  There may potentially be some other context this fragment is used in where sudo isn't appropriate, but this makes it easier to copy/paste the command and have it work on modern linuxes.

itsmylife committed:
InfluxDB: Fix retention policy handling for InfluxDB 3.0 engine by bringing back the hardcoded default (#72467)

* Revert retention policy handling

* Remove comment

* Send retention policy with request when it's possible

eleijonmarck committed:
Auth: OAuth sets `skip_org_role_sync = true` for auth.google by default (#72819)

* sets skip_org_role_sync to true for google

* add google skiporgrolesync and sets to true always

* add field

* Update docs/sources/setup-grafana/configure-security/configure-authentication/google/index.md

Co-authored-by: Christopher Moyer <35463610+chri2547@users.noreply.github.com>

* add AKS to words

* script back to mina

---------

Co-authored-by: Christopher Moyer <35463610+chri2547@users.noreply.github.com>

linoman committed:
Auth: Add documentation references to No Basic role (#72880)

* Update the documentation

* Add references to the documentation

* Replace X with YES in roles table

konrad147 committed:
Alerting: Fix cloud rules editing (#72927)

itsmylife committed:
Chore: Use the const instead of hardcoded string (#72925)

* Use the const instead of hardcoded string

* Use the const instead of hardcoded string

* Fix unit test

drew08t committed:
Geomap: Fix fixed edge text for network layer (#72863)

Geomap: Fix fixed edge text

nmarrs committed:
Trend: Fix x field matching logic (#72862)

yuri-tceretian committed:
Alerting: Refactor of state manager tests (#72849)

* calculate cacheID instead of literals
   * use mocked clocks
   * advance clocks with the eval results
   * use clearer timestamp aliases
   * make expected state labels be more clear to read
Co-authored-by: Matthew Jacobson <matthew.jacobson@grafana.com>

sarahzinger committed:
Documentation: Clarify how to use prometheus locally (#72846)

Co-authored-by: Fiona Artiaga <89225282+GrafanaWriter@users.noreply.github.com>

sarahzinger committed:
Config: making [aws][external_id] part of env (#72062)

* Config: making [aws][external_id] part of env

* Fix go.sum

* Add a test

mrevutskyi committed:
Auth: prevent auto_login redirect if user is already authenticated (#72477)

* Auth: prevent auto_login redirect if user is already authenticated

Before attempting an auto-login for OAuth, verifies if current context has already been
authenticated.

Fixes: #72476

Co-authored-by: Karl Persson <kalle.persson92@gmail.com>

ashharrison90 committed:
Chore: improve typings (#72679)

improve typings

fabrizio-grafana committed:
Fix typos in contributing documentation (#72719)

LudoVio committed:
Transformation: keep collapsed states (#72601)

* Transformation: keep collapsed states

* Update

sarahzinger committed:
Datasource Plugins: Allow tracking for configuration usage (#72650)

Datasource Plugins: Allow tracking for configuration usage

itsmylife committed:
Prometheus: Fix fetching label values when datasource has no labels match api support (#72960)

* interpolate match string

* provide unit test

* add the third parameter back to fix the unit test

Ukochka committed:
sign-plugin-docs: information of usage of a new token has been added to the docs (#72912)

* Documentation for signing a plugin has been updated

* changes after review

* small review change

matyax committed:
LogRowMessage: remove hardcoded true from condition (#72976)

gillesdemey committed:
Alerting: Fix "see graph button" for cloud rules (#72980)

Co-authored-by: Gilles De Mey <gilles.de.mey@gmail.com>
Co-authored-by: Giordano Ricci <me@giordanoricci.com>

ugtar committed:
Plugins: opentsdb ds: interpolate filter variables in expressions (#71232)

Use the same code path to interpolate both metric variables and
filter/tag variables when the data source is accessed either directly or
via the expressions datasource.

partial fix for #48141 (opentsdb datasource only)

Signed-off-by: Uri Okrent <uokrent@gmail.com>

arunkg2787 committed:
docs: add default value workaround (#72545)

* Update _index.md

Fixes #https://github.com/grafana/support-escalations/issues/6376

Variables : Added Default values

* Update docs/sources/dashboards/variables/_index.md

Co-authored-by: Isabel <76437239+imatwawana@users.noreply.github.com>

* Ran prettier to resolve build errors

---------

Co-authored-by: Isabel <76437239+imatwawana@users.noreply.github.com>
Co-authored-by: Isabel Matwawana <isabel.matwawana@grafana.com>
Co-authored-by: Jack Baldry <jack.baldry@grafana.com>

mikkancso committed:
EditDatasource: Fix navtree when `dataSourcePageHeader` is on (#72827)

use useDataSourceSettingsNav from connections in DataSourceTabPage

xnyo committed:
Chore: Bump github.com/grafana/grafana-plugin-sdk-go to v0.172.0 (#72989)

* Bump github.com/grafana/grafana-plugin-sdk-go

* go mod tidy

* go mod: added missing owners

jackw committed:
Chore: Bump semver npm package to latest (#72983)

build(npm): bump semver

svennergr committed:
Loki: Fix wrong context being passed to HTTP client (#72991)

use new context containing tracing information

Ukochka committed:
Docs: Update screenshots for generating an access token for signing a plugin (#73031)

updated screenshots path and name

svennergr committed:
Logs: Fix displaying the wrong field as body (#73025)

* fix displaying the wrong field as body

* fix test

* fix `getFirstFieldOfType` with non-present type

itsmylife committed:
InfluxDB: Fix handling flux response with no time and value column (#72833)

* Upgrade the influxdb-client-go version

* Handle flux response with no time and value column

LudoVio committed:
Transformations: Fix partitionByValues when there is no match (#72981)

JohnnyQQQQ committed:
Alerting: Optimize external Loki queries (#73014)

fauust committed:
Docs: make it clear that MariaDB is also supported as datasource (#71832)

* Make it clear that MariaDB is also supported as datasource

MariaDB is now the default relational DB on lots of Linux distribution,
it's important to make it clear that it is perfectly supported.

* Fix 404 and make MySQL docs URL more robust

And lint with prettier.

* Review from lwandz13

yuri-tceretian committed:
Alerting: Export contact points to check access control action instead legacy role (#71990)

* introduce a new action "alert.provisioning.secrets:read" and role "fixed:alerting.provisioning.secrets:reader"
* update alerting API authorization layer to let the user read provisioning with the new action
* let new action use decrypt flag
* add action and role to docs

aocenas committed:
Table: Add storybook and docs for custom render cell type  (#72619)

* Add storybook and docs for custom rendering cell

* Update packages/grafana-ui/src/components/Table/Table.mdx

Co-authored-by: Victor Marin <36818606+mdvictor@users.noreply.github.com>

---------

Co-authored-by: Victor Marin <36818606+mdvictor@users.noreply.github.com>

konrad147 committed:
Alerting: Optimize rule details page data fetching (#72977)

clementduveau committed:
Add a warning in the documentation (#72975)

A partner complained in the name of a customer. They have 6k datasources and it took them some time to figure out why they only had 5k coming back.

This commit add a warning for this edge case, content is pretty clear.

kalleep committed:
Auth: Use authn.Service for all tests (#72921)

* Dashboards: Fix tests when authn broker is enabled.
StarService was not configured for tests, the call was guarded by !c.IsSignedIn

* Change default to be anon user to match expectations from tests

* OAuth: rewrite tests to work with authn.Service

* Setup template renderer by default

* Extract cookie options from cfg instead of relying on global variables

* Fix test to work with authn service

* Middleware: rewrite auth tests

* Remvoe session cookie if we cannot refresh access token

Jguer committed:
Auth: Add SignedIn user interface NamespacedID (#72944)

* wip

* scope active user to 1 org

* remove TODOs

* add render auth namespace

* import cycle fix

* make condition more readable

* convert Evaluate to user Requester

* only use active OrgID for SearchUserPermissions

* add cache key to interface definition

* change final SignedInUsers to interface

* fix api key managed roles fetch

* fix anon auth id parsing

* Update pkg/services/accesscontrol/acimpl/accesscontrol.go

Co-authored-by: Ieva <ieva.vasiljeva@grafana.com>

---------

Co-authored-by: Ieva <ieva.vasiljeva@grafana.com>

Jguer committed:
Chore: Fix ngalert Evaluate signature change (#73084)

fix ngalert Evaluate sig change

svennergr committed:
Logs: Disable query splitting due to conflicts with transformations (#73086)

* disable query splitting for now

* change stage of `lokiQuerySplitting`

Jguer committed:
Auth: Org Invite and Team API SignedInUser interfacing (#73085)

* fix ngalert Evaluate sig change

* interface for teams and org invites

* Update pkg/api/org_invite.go

Co-authored-by: Ieva <ieva.vasiljeva@grafana.com>

---------

Co-authored-by: Ieva <ieva.vasiljeva@grafana.com>

kalleep committed:
Contexthandler: Remove code that is no longer used (#73101)

* Contexthandler: remove dead code

* Contexthandler: Add tests

* Update pkg/tests/api/alerting/api_alertmanager_test.go

Co-authored-by: Jo <joao.guerreiro@grafana.com>

---------

Co-authored-by: Jo <joao.guerreiro@grafana.com>

gillesdemey committed:
Alerting: Add "no-rules" to redirect to rule viewer (#73111)

mmandrus committed:
Feature Toggles: Create API for updating feature toggle state from the feature toggle admin page  (#73022)

* create roles for writing feature toggles

* create update endpoint / handler

* api changes

* add feature toggle validations

* hide toggles based on their state

* make FlagFeatureToggle read only

* add username log

* add username string

* refactor for better readability

* refactor unit tests so we can do more validations

* some skeletoning for the set tests

* write unit tests for updater

* break helper functions out

* update sample ini to match defaults

* add more logic to ReadOnly label

* add user documentation

* fix lint issue

* Update docs/sources/setup-grafana/configure-grafana/_index.md

Co-authored-by: J Stickler <julie.stickler@grafana.com>

* Update docs/sources/setup-grafana/configure-grafana/_index.md

Co-authored-by: J Stickler <julie.stickler@grafana.com>

* Update docs/sources/setup-grafana/configure-grafana/_index.md

Co-authored-by: J Stickler <julie.stickler@grafana.com>

* Update docs/sources/setup-grafana/configure-grafana/_index.md

Co-authored-by: J Stickler <julie.stickler@grafana.com>

* Update docs/sources/setup-grafana/configure-grafana/_index.md

Co-authored-by: J Stickler <julie.stickler@grafana.com>

* Update docs/sources/setup-grafana/configure-grafana/_index.md

Co-authored-by: J Stickler <julie.stickler@grafana.com>

---------

Co-authored-by: IbrahimCSAE <ibrahim.mdev@gmail.com>
Co-authored-by: J Stickler <julie.stickler@grafana.com>

arunkg2787 committed:
Update _index.md (#72562)

Fixes # https://github.com/grafana/support-escalations/issues/6887

Configure HA: Added ha_peer_Timeout global config in the unified alerting

yuri-tceretian committed:
Alerting: Add more tests for state manager ProcessEvalResults (#73019)

Co-authored-by: Matthew Jacobson <matthew.jacobson@grafana.com>

wbrowne committed:
Plugins: Add validation stage to plugin loader pipeline (#73053)

* first pass

* change validation signature

* err tracking

* fix

* undo golden

* 1 more

* fix

* adjust doc

* add test helper

* fix linter

JohnnyQQQQ committed:
Alerting: optimize rules with multiple loki range queries (#73103)

josmperez committed:
Docs: Fix broken links to create-plugin docs (#72965)

* Fix broken links to create-plugin docs

* Typo fix

* Fix additional broken link

* Fix additional links

kalleep committed:
Auth: Remove auth broker flag and clean up login handlers (#73109)

* Auth: Remove auth broker flag and clean up login handlers

wbrowne committed:
Plugins: Add context to StaticRouteResolver and ErrorResolver interfaces (#73121)

* add ctx

* fix tests

kalleep committed:
Auth: Remove unused Authenticator service (#73143)

Auth: remove unused Authenticator service

joey-grafana committed:
Table: Support display of multiple sub tables (#71953)

* Add nested option to DataFrame. Refactor Table to use nested dataframes for sub-tables

* Use nested frames for TraceQL response

* debugging

* Fix cell text and table position

* Update getItemSize

* noHeader size

* Update sub table renderer

* Update table container height

* Cleanup and fix RawPrometheusContainer height

* Update resultTransformer and docker script

* Updates to TableContainer, resultTransformer after merge

* Fixes for table pagination in dashboards

* Cell height and show footer enhancement/fix

* Sub table links

* Update RawPrometheusContainer

* Remove console log

* Update tests

* Update storybook

* Remove Tempo demo

* Store nested data in single field via its values

* Move nested prop into custom

* Tempo demo

* Add field type & update incorrect logic

* Update docker compose image for Tempo

* Update packages/grafana-data/src/field/fieldOverrides.ts

Co-authored-by: Piotr Jamróz <pm.jamroz@gmail.com>

* Simplify logic for getting nestedFrames and rendering sub tables

* Update docs for table

* Update nested table bg color

* Lighten nested table bg color

* Renames

* Migrate frames using parentRowIndex and add deprecation notice

* Update title

* Align expander icon size between Table and interactive table

* Table: Refactor out the expanded rows bits

* fix spacing

* Add line along left side for expanded rows

* Disable hover row background when expanded

---------

Co-authored-by: André Pereira <adrapereira@gmail.com>
Co-authored-by: Piotr Jamróz <pm.jamroz@gmail.com>
Co-authored-by: Torkel Ödegaard <torkel@grafana.com>

Jguer committed:
Auth: Move Service Account service to SignedInUser Interface (#73142)

* move service account service to identity interface

* Update pkg/services/auth/identity/requester.go

wbrowne committed:
Plugins: Do not fail bootstrap stage if single decorate step fails (#73147)

* don't fail all if decorate step fails

* fix casing

* include err too

* cover pluginsintegration too

stephaniehingtgen committed:
Dev docs: Add a doc around feature toggles (#72938)

axelavargas committed:
Dashboards: PanelChrome -  remove untitled placeholder and add border when panel is transparent (#73150)

Jguer committed:
Auth: Move star service to identity interface (#73114)

* move star service to identity interface

* Update pkg/services/auth/identity/requester.go

Co-authored-by: Ieva <ieva.vasiljeva@grafana.com>

* Update pkg/services/contexthandler/model/model.go

---------

Co-authored-by: Ieva <ieva.vasiljeva@grafana.com>

alyssajoyner committed:
Cloud Monitoring: Support AliasBy property in MQL mode (#73116)

jsartin513 committed:
Docs: Update to include keycloak link in oauth docs (#72698)

Update to include keycloak link in oauth docs

kevinburkesegment committed:
Update index.md (#73077)

DanCech committed:
Slug: Combine various slugify fixes for special character handling (#73164)

* combine various slugify fixes for special character handling

* a couple more test cases

* update more tests

* goimports

Ptoricanmilt committed:
Update [analytics] section to include the ability to set to false (#71473)

* Update [analytics] section to include the ability to set to false

In the follow section: 

https://grafana.com/docs/grafana/latest/setup-grafana/configure-grafana/#analytics

Added a line for 'enabled' under the [analytics] section to include the ability to set to false as this is now a togglable feature.

* Update docs/sources/setup-grafana/configure-grafana/_index.md

* Update docs/sources/setup-grafana/configure-grafana/_index.md

---------

Co-authored-by: Christopher Moyer <35463610+chri2547@users.noreply.github.com>
Co-authored-by: Jack Baldry <jack.baldry@grafana.com>

gillesdemey committed:
Alerting: Always invalidate the AM config after mutation (#73162)

academo committed:
Sandbox: Fix monaco editor custom languages not working correctly inside the sandbox (#72911)

adrapereira committed:
Tempo: Integrate context aware autocomplete API (#67845)

* Send query in search tag values call

* Make sure to send the full query when using the code editor

* Fix merge conflicts

* Remove unused params

sympatheticmoose committed:
docs: remove strava from angular plugin list (#73192)

JohnnyQQQQ committed:
docs(alerting): add section about running redis for HA (#73153)

aangelisc committed:
AzureMonitor: Allow `serviceTags` and `tags` to be empty for trace results (#73145)

Allow serviceTags and tags to be empty

kevinburkesegment committed:
Update index.md (#73183)

* Update index.md

* Update docs/sources/setup-grafana/configure-security/configure-authentication/okta/index.md

---------

Co-authored-by: Christopher Moyer <35463610+chri2547@users.noreply.github.com>

sympatheticmoose committed:
docs: remove Moogsoft from angular list, update version for helm (#73204)

VikaCep committed:
Alerting: Remove dump wrapper for yaml config (#73169)

Remove dump wrapper for yaml config

ivanahuckova committed:
Logs: Improve plugins developer documentation for logs APIs (#72441)

* Logs: Add plugins developer documentation for logs APIs

* Update

* Update

* Update docs/sources/developers/plugins/create-a-grafana-plugin/extend-a-plugin/add-log-queries.md

Co-authored-by: Matias Chomicki <matyax@gmail.com>

* Update docs/sources/developers/plugins/create-a-grafana-plugin/extend-a-plugin/add-log-queries.md

Co-authored-by: Matias Chomicki <matyax@gmail.com>

* Update docs/sources/developers/plugins/create-a-grafana-plugin/extend-a-plugin/add-log-queries.md

Co-authored-by: Matias Chomicki <matyax@gmail.com>

* Update docs/sources/developers/plugins/create-a-grafana-plugin/extend-a-plugin/add-log-queries.md

Co-authored-by: Matias Chomicki <matyax@gmail.com>

* Update docs/sources/developers/plugins/create-a-grafana-plugin/extend-a-plugin/add-log-queries.md

Co-authored-by: Matias Chomicki <matyax@gmail.com>

* Update docs/sources/developers/plugins/create-a-grafana-plugin/extend-a-plugin/add-log-queries.md

Co-authored-by: Matias Chomicki <matyax@gmail.com>

* Update docs/sources/developers/plugins/create-a-grafana-plugin/extend-a-plugin/add-log-queries.md

Co-authored-by: Matias Chomicki <matyax@gmail.com>

* Update docs/sources/developers/plugins/create-a-grafana-plugin/extend-a-plugin/add-log-queries.md

Co-authored-by: Matias Chomicki <matyax@gmail.com>

* Update docs/sources/developers/plugins/create-a-grafana-plugin/extend-a-plugin/add-log-queries.md

Co-authored-by: Matias Chomicki <matyax@gmail.com>

* Update docs/sources/developers/plugins/create-a-grafana-plugin/extend-a-plugin/add-log-queries.md

Co-authored-by: Matias Chomicki <matyax@gmail.com>

* Update docs/sources/developers/plugins/create-a-grafana-plugin/extend-a-plugin/add-log-queries.md

* Update docs/sources/developers/plugins/create-a-grafana-plugin/extend-a-plugin/add-log-queries.md

Co-authored-by: Haris Rozajac <58232930+harisrozajac@users.noreply.github.com>

* Update docs/sources/developers/plugins/create-a-grafana-plugin/extend-a-plugin/add-log-queries.md

Co-authored-by: Matias Chomicki <matyax@gmail.com>

* Update docs/sources/developers/plugins/create-a-grafana-plugin/extend-a-plugin/add-log-queries.md

Co-authored-by: Piotr Jamróz <pm.jamroz@gmail.com>

* Update docs/sources/developers/plugins/create-a-grafana-plugin/extend-a-plugin/add-log-queries.md

Co-authored-by: Matias Chomicki <matyax@gmail.com>

* Update docs/sources/developers/plugins/create-a-grafana-plugin/extend-a-plugin/add-log-queries.md

Co-authored-by: Matias Chomicki <matyax@gmail.com>

* Update docs/sources/developers/plugins/create-a-grafana-plugin/extend-a-plugin/add-log-queries.md

Co-authored-by: Matias Chomicki <matyax@gmail.com>

* Update docs/sources/developers/plugins/create-a-grafana-plugin/extend-a-plugin/add-log-queries.md

Co-authored-by: Matias Chomicki <matyax@gmail.com>

* Update docs/sources/developers/plugins/create-a-grafana-plugin/extend-a-plugin/add-log-queries.md

Co-authored-by: Matias Chomicki <matyax@gmail.com>

* Update docs/sources/developers/plugins/create-a-grafana-plugin/extend-a-plugin/add-log-queries.md

Co-authored-by: Matias Chomicki <matyax@gmail.com>

* Update docs/sources/developers/plugins/create-a-grafana-plugin/extend-a-plugin/add-log-queries.md

Co-authored-by: Matias Chomicki <matyax@gmail.com>

* Update docs/sources/developers/plugins/create-a-grafana-plugin/extend-a-plugin/add-log-queries.md

Co-authored-by: Matias Chomicki <matyax@gmail.com>

* Update docs/sources/developers/plugins/create-a-grafana-plugin/extend-a-plugin/add-log-queries.md

* Update docs/sources/developers/plugins/create-a-grafana-plugin/extend-a-plugin/add-log-queries.md

Co-authored-by: Matias Chomicki <matyax@gmail.com>

* Update docs/sources/developers/plugins/create-a-grafana-plugin/extend-a-plugin/add-log-queries.md

Co-authored-by: Matias Chomicki <matyax@gmail.com>

* Update based on feedback

* Update common labels content

* Remove redundant comment notes

* Adjust code snippets

* Move documentation into ow to build a logs data source plugin

* Propose edits for enhancements to logs datasource plugins tutorial (#73074)

* Initial commiit

* Additional edits

* Additional edits

* Update docs/sources/developers/plugins/create-a-grafana-plugin/develop-a-plugin/build-a-logs-data-source-plugin.md

* Update docs/sources/developers/plugins/create-a-grafana-plugin/develop-a-plugin/build-a-logs-data-source-plugin.md

* Add required field info into a table

---------

Co-authored-by: Matias Chomicki <matyax@gmail.com>
Co-authored-by: Haris Rozajac <58232930+harisrozajac@users.noreply.github.com>
Co-authored-by: Piotr Jamróz <pm.jamroz@gmail.com>
Co-authored-by: Joseph Perez <45749060+josmperez@users.noreply.github.com>

fabrizio-grafana committed:
Fix error message for variable names (#72580)

asimonok committed:
Drawer: Fix scrolling drawer content on Safari (#73080)

Add min-height to allow scroll drawer content on Safari

kalleep committed:
RBAC: remove LoadPermissionsMiddleware (#73228)

* PubDash: remove LoadPermissionMiddleware from tests

* RBAC: Remove unused LoadPermission middleware

wbrowne committed:
Plugins: Remove unnecessary error result from env vars interface (#73224)

* remove error from interface

* fix test

joey-grafana committed:
Tracing: Split name column in search results (#72449)

* Split name column

* Add tests

* Update test

matyax committed:
Loki: Migrate HTTP settings to new components (#72831)

* Loki: migrate data source http settings

* Remove unused export

* Update e2e tests

* Update unit test

ashharrison90 committed:
Chore: fix some more types (#72726)

fix some more types

ivanahuckova committed:
Elasticsearch: Enable running of queries trough data source backend (#73222)

Elasticsearch: Enable enableElasticsearchBackendQuerying feature toggle by default

sarahzinger committed:
Cloudwatch: reorg files in components (#73176)

sarahzinger committed:
Cloudwatch: Upgrade aws-sdk and display external ids for temporary credentials (#72821)

(under a feature toggle, not yet ready for public testing)

gtk-grafana committed:
InfluxDB: make influxql options the default if nothing defined (#73247)

make influxql options the default if nothing defined

ashharrison90 committed:
CodeEditor: Correctly fires onChange handler (#73030)

move onChange outside onEditorDidMount condition

svennergr committed:
Logs: Fix scrolling with `exploreScrollableLogsContainer` feature (#73272)

fix scrolling with `exploreScrollableLogsContainer`

alexanderzobnin committed:
Geomap: Fix crosshair glitch (#72909)

kaydelaney committed:
Dashboard: Fix repeated row panel placement with larger number of rows (#72011)

Closes #69833

lwandz13 committed:
docs: What’s New & Upgrade Guide 10.1 (#70636)

* added what's new 10.1

* created new upgrade guide

* updated index file to 10.1

* updated whats new index file

* Dataviz: add disconnect values whats new content

* Docs: Add Configure refresh token handling separately for OAuth providers to what's new 10.1 (#71648)

* Add what s new section for configuring refresh token handling

---------

Co-authored-by: linoman <2051016+linoman@users.noreply.github.com>
Co-authored-by: Isabel <76437239+imatwawana@users.noreply.github.com>

* Whats new: Add step editor and copy link  (#71964)

* Add step editor and copu link to the whats new

* Update docs/sources/whatsnew/whats-new-in-v10-1.md

Co-authored-by: Isabel <76437239+imatwawana@users.noreply.github.com>

* Update docs/sources/whatsnew/whats-new-in-v10-1.md

Co-authored-by: Isabel <76437239+imatwawana@users.noreply.github.com>

* Update docs/sources/whatsnew/whats-new-in-v10-1.md

Co-authored-by: Gábor Farkas <gabor.farkas@gmail.com>

---------

Co-authored-by: Isabel <76437239+imatwawana@users.noreply.github.com>
Co-authored-by: Gábor Farkas <gabor.farkas@gmail.com>

* Add flamegraph section

* Added TraceQL response streaming section

* Updated tempo-streaming video link

* What's new: Add Loki, Elasticsearch, and Log changes (#72101)

* What's new: Add Loki, Elasticsearch, and Log changes

* Remove Lucene parser mention

* added contributor name to each entry

* Alerting: adds whats new entries and structure

* Elasticsearch: logs sample in explore

* Add span filtering is GA to whats new doc

* Transformations redesign and metrics explorer

* ran prettier

* What's new 10.1: Recategorize contents and add missing data (#72199)

What's new: recategorize content and add missing data

* Revert whats new refresh token handling (#72191)

Revert refresh token handling

* docs: What’s New & Upgrade Guide 10.1 AuthNZ (#72000)

* add what's new authnz

* Update docs/sources/upgrade-guide/upgrade-v10.1/index.md

---------

Co-authored-by: Ieva <ieva.vasiljeva@grafana.com>
Co-authored-by: linoman <2051016+linoman@users.noreply.github.com>
Co-authored-by: Isabel <76437239+imatwawana@users.noreply.github.com>

* Alerting: screenshots and updates

* Alerting: delete couple of entries

* Docs: Add Visualizations and Widgets Split to what's new 10.1 (#72009)

Co-authored-by: Isabel <76437239+imatwawana@users.noreply.github.com>

* Docs: Add CloudWatch Logs Monaco query editor to what's new in 10.1 (#72221)

* Docs: Add CloudWatch Logs Monaco query editor to what's new in 10.1

* Add guidance for Cloud users to enable the feature

Co-authored-by: Isabel <76437239+imatwawana@users.noreply.github.com>

---------

Co-authored-by: Isabel <76437239+imatwawana@users.noreply.github.com>

* Alerting: adds notification policy entry

* Add Plugins/Angular deprecation sections to What's new 10.1

* Add Plugins/ds uid deprecation section to What's new 10.1

* Alerting: adds label color entry

* Adds whats new features for AWS Datasources

* Update what's new 10.1 angular deprecation section

* Docs: what's new for preventing manual role updates (#72279)

* what's new for disabling manual role updates

* add a reference to the feature toggle

* fix spelling

* Apply suggestions from code review

Co-authored-by: Gabriel MABILLE <gamab@users.noreply.github.com>

* PR feedback

---------

Co-authored-by: Gabriel MABILLE <gamab@users.noreply.github.com>

* wording fixes

* whatsnew-10.1.0: add logs-context info (#72240)

* whatsnew-10.1.0: add logs-context info

* explain what log context is

* improved description

Co-authored-by: Ivana Huckova <30407135+ivanahuckova@users.noreply.github.com>

* better phrasing

---------

Co-authored-by: Ivana Huckova <30407135+ivanahuckova@users.noreply.github.com>

* Subfolders: folder picker

* Influxdb backend mode

* Dataviz: heatmap datalink support

* DataViz: geomap new beta network layer

* Dataviz: update geomap network layer image

* Dataviz: update availability wording to public preview from experimental for geomap network layer

* Docs: Add Activate draft reports section to what's new in 10.1

* Docs: Add format time transformation to what's new

* Docs: Adjust format time transformation language

* Explore: Add what's new about panel plugins (#72338)

* Add what's new about panel plugins

* Update docs/sources/whatsnew/whats-new-in-v10-1.md

Co-authored-by: Isabel <76437239+imatwawana@users.noreply.github.com>

---------

Co-authored-by: Isabel <76437239+imatwawana@users.noreply.github.com>

* add tabular outer join to whats new

* Update traceql streaming section

* Added new tempo search section

* Update traceql streaming video name

* docs: what's new 10.1.0, easier to use elastic query editor (#72513)

* docs: elasticsearch: new mode selector

* improved text

Co-authored-by: Ivana Huckova <30407135+ivanahuckova@users.noreply.github.com>

---------

Co-authored-by: Ivana Huckova <30407135+ivanahuckova@users.noreply.github.com>

* mitch edits

* Added product labels

* Made general copy edits

* Removed editing notes

* Fixed broken relref

* Fixed typo and added contributor name

* Added contributor name

* Made minor copy edits

* Update format time transformation docs

* A small edit to the role list

* Added missing information and committed copy edits

* Move Reports item to under Dashboards & visualizations

* Changed heading level of Activate draft reports

* Fixed wording of Choose which fields to display in log line entry

* Made edits to Logs improved performance entry

* Added availability for Transformations redesign

* Added availability for Join by fields

* Made final copy edits

* Removed Loki query splitting

---------

Co-authored-by: nmarrs <nathanielmarrs@gmail.com>
Co-authored-by: Misi <mgyongyosi@users.noreply.github.com>
Co-authored-by: linoman <2051016+linoman@users.noreply.github.com>
Co-authored-by: Isabel <76437239+imatwawana@users.noreply.github.com>
Co-authored-by: Ivana Huckova <30407135+ivanahuckova@users.noreply.github.com>
Co-authored-by: Gábor Farkas <gabor.farkas@gmail.com>
Co-authored-by: Andrej Ocenas <mr.ocenas@gmail.com>
Co-authored-by: André Pereira <adrapereira@gmail.com>
Co-authored-by: Matias Chomicki <matyax@gmail.com>
Co-authored-by: brendamuir <100768211+brendamuir@users.noreply.github.com>
Co-authored-by: Gareth Dawson <gwdawson.work@gmail.com>
Co-authored-by: Joey Tawadrous <joey.tawadrous@grafana.com>
Co-authored-by: catherineymgui <catherine.gui@grafana.com>
Co-authored-by: Jo <joao.guerreiro@grafana.com>
Co-authored-by: Ieva <ieva.vasiljeva@grafana.com>
Co-authored-by: Alexa V <239999+axelavargas@users.noreply.github.com>
Co-authored-by: Kevin Yu <kevinwcyu@users.noreply.github.com>
Co-authored-by: Giuseppe Guerra <giuseppe@guerra.in>
Co-authored-by: Sarah Zinger <sarah.zinger@grafana.com>
Co-authored-by: Gabriel MABILLE <gamab@users.noreply.github.com>
Co-authored-by: Zsofia <zsofia.komaromi@gmail.com>
Co-authored-by: ismail simsek <ismailsimsek09@gmail.com>
Co-authored-by: Robert Horvath <robert.horvath@grafana.com>
Co-authored-by: Kyle Cunningham <kyle@codeincarnate.com>
Co-authored-by: Piotr Jamróz <pm.jamroz@gmail.com>
Co-authored-by: bohandley <brendan.ohandley@gmail.com>
Co-authored-by: Mitchel Seaman <mitchel.seaman@gmail.com>
Co-authored-by: Isabel Matwawana <isabel.matwawana@grafana.com>
Co-authored-by: Jack Baldry <jack.baldry@grafana.com>

ciancullinan committed:
CloudWatch: Add missing AppFlow metrics (#73149)

IevaVasiljeva committed:
Docs: correct SAML docs (#73281)

Correct SAML docs

yuri-tceretian committed:
Alerting: Update state manager to change all current states in the case when Error\NoData is executed as Ok\Nomal (#68142)

ivanahuckova committed:
Docs: Add plugin dev documentation for logs to trace (#73225)

* Docs: Add documentation for logs to trace

* Update

* Remove space

* Update docs/sources/developers/plugins/create-a-grafana-plugin/develop-a-plugin/build-a-logs-data-source-plugin.md

Co-authored-by: Sven Grossmann <sven.grossmann@grafana.com>

* Replace deprecated MutableDataFrame with createDataFrame

* Update docs/sources/developers/plugins/create-a-grafana-plugin/develop-a-plugin/build-a-logs-data-source-plugin.md

Co-authored-by: Sven Grossmann <sven.grossmann@grafana.com>

* Update docs/sources/developers/plugins/create-a-grafana-plugin/develop-a-plugin/build-a-logs-data-source-plugin.md

Co-authored-by: Joseph Perez <45749060+josmperez@users.noreply.github.com>

* Update docs/sources/developers/plugins/create-a-grafana-plugin/develop-a-plugin/build-a-logs-data-source-plugin.md

Co-authored-by: Joseph Perez <45749060+josmperez@users.noreply.github.com>

---------

Co-authored-by: Sven Grossmann <sven.grossmann@grafana.com>
Co-authored-by: Joseph Perez <45749060+josmperez@users.noreply.github.com>

gtk-grafana committed:
InfluxDB: Tests for #73247 (#73250)

* add test assertions to cover default config UI state in influxDB plugin

nhudson committed:
PostgreSQL: bump lib/pq to latest version (#72416)

* bump lib/pq to latest version

juanicabanas committed:
PublicDashboards: Query order bug fixed (#73293)

gtk-grafana commented:
I love the functionality of letting folks define which labels to display in the exemplar
<img width="1728" alt="image" src="https://github.com/grafana/grafana/assets/109082771/9e6cb96f-bdb3-45c1-a058-2e337cc99c2f">

But there is currently a bug on this branch for exemplars in a heatmap viz which is throwing an error saying that ` tags.includes is not a function`

<img width="1728" alt="image" src="https://github.com/grafana/grafana/assets/109082771/e0129c30-e4c3-45a4-ae4e-d82bfa25e2fa">

it looks like it's expecting an array for the `tags` variable, but it's being passed an object containing the color for the exemplar, but only in the heatmap panel.

Also it might be a good idea to compress the exemplar settings into a single collapsable section, instead of Exemplars, and Exemplars Configuration.

iwysiu committed:
CloudWatch Logs: Wrap sync error from executeGetQueryResults (#73252)

yuri-tceretian committed:
Alerting: Update Discord settings to treat 'url' as a secure setting (#69588)

* make discord url secure

* support migrating unsecure settings to secure settings

* Update public/app/features/alerting/unified/utils/receiver-form.ts

Co-authored-by: William Wernert <william.wernert@grafana.com>

---------

Co-authored-by: Gilles De Mey <gilles.de.mey@gmail.com>
Co-authored-by: William Wernert <william.wernert@grafana.com>

yuri-tceretian committed:
Alerting: Add new metrics and tracings to state manager and scheduler (#71398)

* add metrics and tracing to state manager

* propagate tracer to state manager

* add scheduler metrics

* fix backtesting

* add test for state metrics

* remove StateUpdateCount

* update docs

* metrics can be null

* add tracer to new tests

mgyongyosi committed:
RBAC: Remove unused scope from alert.instances:read fixed role (#73268)

Fix alert.instances:read scope for fixed role

petethepig committed:
Navigation: Remove "preview" from Pyroscope Profiles nav item (#73070)

* Pyroscope: Mark profiles as GA (removes "preview")

* update

wbrowne committed:
Plugins: Remove registry dependency from process manager (#73241)

simplify

linoman committed:
Auth: Add empty role usage metrics for service and user accounts (#73108)

* Add tests for service accounts metrics usage

* Add service account store implementation

* Add service account service implementation

* Add tests for org metrics usage

* Add org implementation

* Add service implementation

joshhunt committed:
Chore: Specify CGO_ENABLED for bingo-install drone in the correct place (#73290)

ricci2511 committed:
DataSourceProxy: Fix url validation error handling (#73087)

* DataSourceProxy: Fix url validation error handling

* DataSourceProxy: Add unit test for proxyDatasourceRequest

* DataSourceProxy: Fix lint error

bergquist committed:
Middleware: Add team metadata to HTTP handlers (#71010)

Signed-off-by: bergquist <carl.bergquist@gmail.com>

wbrowne committed:
Plugins: Make Installer responsible for removing plugins from file system (#73323)

* installer is responsible for removing from file system

* take plugin as arg

* remove resolve step

* return plugin in test

gillesdemey committed:
Email: Light theme email templates (#72398)

axelavargas committed:
Dashboard: Select the last used data source by default when adding a panel to a dashboard (#71777)

Co-authored-by: Juan Cabanas <juan.cabanas@grafana.com>

Yelmor committed:
Dashboard: Variables Dropdown - Fix Flickering on Refresh (#72566)

matyax committed:
Log row: split row-highlighting booleans in two (#73330)

svennergr committed:
Loki: Fix validation of `step` values to also allow e.g. `ms` values (#73270)

* use go duration validation

* add `isValidGrafanaDuration`

* use `isValidGrafanaDuration`

* improve jsdoc

adela-almasan committed:
Canvas: Fix data links (#73332)

aishyandapalli commented:
@gtk-grafana Thanks for looking into this. I will address the above mentioned issues and update the PR

glitchwizard committed:
Docs: Fixed Grafana Fundamentals tutorial to match updated UI (#73168)

* Update index.md

The word `Configuration` was used to describe an action for getting started, when the actual UI has the word `Administration` and not Configuration. Updated the docs to reflect the actual environment created.

* Update index.md

Updated the tutorial to match the current UI options

* Made copy edits for styling and consistency with Cloud fundamentals tutorial

Made some copy edits to match our style guide and to make this tutorial a bit more consistent with the Cloud version. Also added the step for updating the legend back to the build dashboards section.

* Ran linter

* Edited text to match updated UI

Made edits to match UI behaviour as well as some further style edits.

* Made final copy edits

---------

Co-authored-by: Isabel <76437239+imatwawana@users.noreply.github.com>
Co-authored-by: Isabel Matwawana <isabel.matwawana@grafana.com>
